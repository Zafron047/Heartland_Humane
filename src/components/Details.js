import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Nav from './Nav';
import { fetchBreed } from '../redux/detailReducer/detailSlice';
import '../styles/Detail.css';

const Details = () => {
  const breedData = useSelector((state) => state.CatBreeds.breedData);
  const loading = useSelector((state) => state.CatBreeds.loading);
  const error = useSelector((state) => state.CatBreeds.error);
  const dispatch = useDispatch();
  const { breedId } = useParams();

  useEffect(() => {
    dispatch(fetchBreed(breedId));
  }, [dispatch, breedId]);

  if (loading) {
    return <p className="loading">Loading...</p>;
  } if (error) {
    return (
      <p>
        Error fetching:
        { error }
      </p>
    );
  } return (
    <div className="container">
      <Nav />
      {breedData.length > 0 && (
      <>
        <h1 className="detail-header">
          Breed:
          {' '}
          {breedData[0].name}
        </h1>
        <div className="grid-container">
          <p className="breed-feature">
            Lifespan:
            {' '}
            {breedData[0].lifeSpan}
            {' '}
            years
          </p>
          <p className="breed-feature">
            Adaptability:
            {' '}
            {breedData[0].adaptability}
            {' '}
            / 5
          </p>
          <p className="breed-feature">
            AffectionLevel:
            {' '}
            {breedData[0].affectionLevel}
            {' '}
            / 5
          </p>
          <p className="breed-feature">
            EnergyLevel:
            {' '}
            {breedData[0].energyLevel}
            {' '}
            / 5
          </p>
          <p className="breed-feature">
            HealthIssues:
            {' '}
            {breedData[0].healthIssues}
            {' '}
            / 5
          </p>
          <p className="breed-feature">
            Intelligence:
            {' '}
            {breedData[0].intelligence}
            {' '}
            / 5
          </p>
        </div>
      </>
      )}
      <ul className="img-container">
        {breedData.map((item) => (
          <li className="image-box" key={item.id}>
            <img className="cat-img" src={item.url} alt="A Cat" />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Details;
