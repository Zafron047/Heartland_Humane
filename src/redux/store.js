import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './homeReducer/HomeSlice';
import detailReducer from './detailReducer/detailSlice';

const store = configureStore({
  reducer: {
    AllKats: homeReducer,
    CatBreeds: detailReducer,
  },
});

export default store;
