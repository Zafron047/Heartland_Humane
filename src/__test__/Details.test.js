import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Details from '../components/Details';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    <MemoryRouter>
      {component}
    </MemoryRouter>
  </Provider>,
);

describe('Details', () => {
  it('render Details Components', () => {
    render(<Details />);
    expect(Details).toMatchSnapshot();
  });
});
