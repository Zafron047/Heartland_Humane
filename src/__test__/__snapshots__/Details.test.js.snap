// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details render Details Components 1`] = `[Function]`;
